src/umpire/strategy/mixins/AlignedAllocation.inl

```diff
@@ -17,28 +17,30 @@ namespace mixins {
 
 inline std::size_t AlignedAllocation::aligned_round_up(std::size_t size)
 {
-    return size + (m_alignment - 1) - (size - 1) % m_alignment;
+  return size + (m_alignment - 1) - (size - 1) % m_alignment;
 }
 
 inline void* AlignedAllocation::aligned_allocate(std::size_t size)
 {
-    std::size_t total_bytes{ size + m_alignment - 1 };
-    uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
+  std::size_t total_bytes{ size + m_alignment };
+  uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
 
-    uintptr_t alignment{ m_alignment - 1 };
-    uintptr_t mask{ m_mask };
-    void *aligned_ptr{ reinterpret_cast<void*>( (ptr + alignment) & mask ) };
+  UMPIRE_POISON_MEMORY_REGION(m_allocator, reinterpret_cast<void*>(ptr), total_bytes);
 
-    base_pointer_map[aligned_ptr] = reinterpret_cast<void*>(ptr);
+  uintptr_t alignment{ m_alignment - 1 };
+  uintptr_t mask{ m_mask };
+  void *aligned_ptr{ reinterpret_cast<void*>( (ptr + alignment) & mask ) };
```

Comment:
c.f. comment on line 26.

Reply:
If the memory is aligned, then the base_pointer and aligned pointer will simply be the same.